src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRateServiceTest.kt

```diff
@@ -408,6 +408,36 @@ class PayRateServiceTest {
       }
     }
 
+    @Test
+    fun `should be able to change a future ended pay rate`() {
+      nomisDataBuilder.build {
+        programService {
+          courseActivity = courseActivity {
+            courseSchedule()
+            courseScheduleRule()
+            payRate(iepLevelCode = "STD", payBandCode = "5", endDate = tomorrow.toString(), halfDayRate = 1.41)
+          }
+        }
+      }
+      val request = listOf(
+        PayRateRequest("STD", "5", BigDecimal(0.1)),
+      )
+      val newPayRates = payRatesService.buildNewPayRates(request, courseActivity)
+
+      assertThat(newPayRates.size).isEqualTo(2)
+      // old rate should now expire today
+      with(newPayRates.findRate("STD", "5", expired = true)) {
+        assertThat(halfDayRate).isCloseTo(BigDecimal(1.41), within(BigDecimal(0.001)))
```

Comment:
this looks a bit odd - why the `closeTo`?  If you called `BigDecimal` constructor with a string instead then you should be able to use [isEqualByComparingTo](https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractBigDecimalAssert.html#isEqualByComparingTo-java.lang.String-)